apps/sms/services/js/message_manager.js

```diff
@@ -396,26 +495,45 @@ var MessageManager = {
     });
   },
 
+  /**
+   * Marks a thread as read or unread depending on value of isRead as true or
+   * false respectively.
+   * @param {number|string} threadId The ID of the thread to be marked as read.
+   * @param {boolean} isRead if "true" then message is to be marked as read
+   * otherwise message is to be marked as unread.
+   */
   markThreadRead: function mm_markThreadRead(threadId, isRead = true) {
+    // Create filter object to mark all unread messages of the specified thread.
     var filter = {};
     filter.threadId = +threadId;
     filter.read = !isRead;
 
     var messagesUnreadIDs = [];
+
+    // Building the options field in getMessages() with above filter.
```

Comment:
Honestly looks like too verbose, maybe we don't need to describe this as it's pretty clear by the code - but up to you